File: python/datastructures/stack/oop_stack.py
class Stack:

    def __init__(self):
        self.items = []


    def push(self,x):
        self.items.append(x)


    def pop(self):
        return self.items.pop()

    def peek(self):
        return self.items[len(self.items)-1]

stack = Stack()

stack.push(1)
print(stack.items)

stack.push(2)
print(stack.peek())
print(stack.items)

print(stack.pop())

File: javascript/datastructures/stack/oop_stack.js
class Stack
{
    constructor()
    {
        this.items = []
    }

    push(item)
    {
        this.items.push(item)
    }

    pop()
    {
       return this.items.pop()
    }

    peek()
    {
        return this.items[this.items.length -1]
    }

}

let stack = new Stack

stack.push(1)
stack.push(2)
stack.push(3)

console.log(stack)

console.log(stack.pop())

console.log(stack.peek())
